node/src/BaseClient.ts

```diff
@@ -1781,23 +1858,39 @@ export class BaseClient {
         return this.createWritePromise(createHExists(key, field));
     }
 
-    /** Returns all fields and values of the hash stored at `key`.
+    /**
+     * Returns all fields and values of the hash stored at `key`.
      *
      * @see {@link https://valkey.io/commands/hgetall/|valkey.io} for details.
      *
      * @param key - The key of the hash.
-     * @returns a list of fields and their values stored in the hash. Every field name in the list is followed by its value.
+     * @param options - (Optional) See {@link DecoderOption}.
+     * @returns A list of fields and their values stored in the hash.
      * If `key` does not exist, it returns an empty list.
      *
      * @example
      * ```typescript
      * // Example usage of the hgetall method
      * const result = await client.hgetall("my_hash");
-     * console.log(result); // Output: {"field1": "value1", "field2": "value2"}
+     * console.log(result); // Output:
+     * // [
+     * //     {"field": "field1", "value": "value1"},
+     * //     {"field", "field2", "value": "value2"}
+     * // ]
      * ```
      */
-    public async hgetall(key: string): Promise<Record<string, string>> {
-        return this.createWritePromise(createHGetAll(key));
+    public async hgetall(
+        key: string,
```

Comment:
This branch is not synced with main? I have this as GlideString in local.

Reply:
This branch updates only return type of these commands

Reply:
yep, you will need to take a pull from main as there are merge conflicts. Plus it will be easier to review with the latest code merged.

Reply:
Merged and fixed, please validate